### PR TITLE
rolls out merkle shreds to ~50% of testnet

### PR DESCRIPTION
**Pull request by behzadnouri**
#### Problem
Continue Merkle shreds rollout off last one: https://github.com/solana-labs/solana/pull/28905

#### Summary of Changes
Roll out merkle shreds to ~50% of testnet

### CODE REVIEW

File: core/src/broadcast_stage/standard_broadcast_run.rs [hunk 1]
```diff
@@ -498,8 +498,8 @@ impl BroadcastRun for StandardBroadcastRun {
     }
 }
 
-fn should_use_merkle_variant(_slot: Slot, _cluster_type: ClusterType, _shred_version: u16) -> bool {
-    false
+fn should_use_merkle_variant(slot: Slot, cluster_type: ClusterType, shred_version: u16) -> bool {
+    cluster_type == ClusterType::Testnet && shred_version == 28353 && (slot % 19 < 10)
```
Comment (steviez) [inline core/src/broadcast_stage/standard_broadcast_run.rs:502]: I guess the idea of checking `shred_version` is that these will automatically get disabled if there is a testnet outage ?

**Review by steviez — COMMENTED** [review]
General timeline / history question as I'm not the most familiar with this feature:
- https://github.com/solana-labs/solana/pull/28905 enabled 20% on testnet
- Above item was going well, but you wanted to make a breaking change so you disabled with https://github.com/solana-labs/solana/pull/29340
- You made the mentioned change, but since haven't had the opportunity to roll out merkle shreds on testnet again?

File: core/src/broadcast_stage/standard_broadcast_run.rs (hunk 1 above)
Comment (behzadnouri) [reply core/src/broadcast_stage/standard_broadcast_run.rs:502]: It is just an extra sanity check that we are on the right cluster.
It does not matter much and I will remove it once we start the roll out on mainnet.

**Comment by behzadnouri** [comment]
> General timeline / history question as I'm not the most familiar with this feature:
> 
>     * [rolls out merkle shreds to ~20% of testnet #28905](https://github.com/solana-labs/solana/pull/28905) enabled 20% on testnet
> 
>     * Above item was going well, but you wanted to make a breaking change so you disabled with [rolls back merkle shreds on testnet #29340](https://github.com/solana-labs/solana/pull/29340)
> 
>     * You made the mentioned change, but since haven't had the opportunity to roll out merkle shreds on testnet again?

yeah, that is accurate.

**Comment by behzadnouri** [comment]
This is the distribution of repair and block propagation time of a ~5 hour window on testnet:
```
2023-06-28T09:41:45 - 2023-06-28T14:41 UTC
```
For `num-repaired / num-data-shreds`:
```
LEGACY                                            MERKLE
nodes  50%     75%     90%     95%     99%        nodes  50%     75%     90%     95%     99%
slots                                             slots
50%    0.0  0.0000  0.0000  0.0000  0.0068        50%    0.0  0.0000  0.0000  0.0000  0.0069
75%    0.0  0.0000  0.0000  0.0000  0.0098        75%    0.0  0.0000  0.0000  0.0000  0.0102
90%    0.0  0.0000  0.0000  0.0007  0.0148        90%    0.0  0.0000  0.0000  0.0000  0.0149
95%    0.0  0.0000  0.0000  0.0029  0.0200        95%    0.0  0.0000  0.0000  0.0023  0.0190
99%    0.0  0.0009  0.0088  0.0152  0.0455        99%    0.0  0.0016  0.0085  0.0133  0.0431
```
Merkle shreds seem better in the tail of distribution (i.e. worst slots/nodes are now better than before).
For block propagation time, `millis/shred`:
```
LEGACY                                            MERKLE
nodes     50%     75%     90%     95%      99%    nodes     50%     75%     90%     95%      99%
slots                                             slots
50%    0.6213  0.6483  0.6709  0.6876   1.0371    50%    0.5874  0.6144  0.6365  0.6531   0.9902
75%    1.0246  1.0920  1.1707  1.2253   1.6482    75%    0.9584  1.0227  1.0896  1.1348   1.5702
90%    2.5455  2.7231  2.8311  2.9069   4.0634    90%    2.3117  2.4552  2.5398  2.6000   3.7299
95%    3.2269  3.4398  3.5741  3.6702   5.1208    95%    2.9193  3.0852  3.1882  3.2690   4.7406
99%    7.7328  8.0086  8.5791  8.9116  11.7288    99%    7.3002  7.6984  8.0088  8.3132  11.4121
```
Rows are slots distribution and columns are nodes distribution.
So `8.9116` in the last rows means that, with legacy shreds, 99% of slots are propagated to 95% of cluster in `< 8.9116` millis per shred, whereas with Merkle shreds this takes `< 8.3132`. So Merkle shreds are `6.7%` better (for this specific point in the distribution).

**Pull request merged**